feat: add ansible host parameter to insights configuration (#155)
linux-system-roles/rhc: README.md

Whether the system automatically updates the dynamic configuration. It is
enabled by default.

```yaml
    rhc_insights:
      ansible_host: "example-host"
```

> Configures the ansible host name with a custom value for the system record

Reviewer: ```suggestion
Configures the Ansible host name with a custom value for the system record
```

> in Host Based Inventory (HBI). This host name is used in playbooks by remediations.

Reviewer: ```suggestion
in Host Based Inventory (HBI). This host name is used in playbooks by Remediations.
```

It is `null` by default and will use the system host name if not specified.

Possible values of this variable:

> * `null` or an empty value: the ansible host name is not changed.

Reviewer: What is an empty value? An empty string? The `omit` 
magic value?

```suggestion
* `null` or an empty value: the Ansible host name is not changed.
```

Reviewer 2: like this:
```yaml
    rhc_insights:
      ansible_host:
```
This is an "empty" value in YAML which is translated to a `null` - see https://yaml.org/spec/1.2.2/#24-tags "Example 2.21 Miscellaneous"

I say "empty" because, although it looks empty, that there is nothing there, in YAML this means to assign the value `null`

Reviewer: I see. I knew about this feature of YAML, it just didn’t come to my mind.

Since the so-called empty value translates to _null_ on the YAML level, it may not be worth mentioning here: the value is still _null_ and the variations of the syntax are property of the YAML format, not Ansible or the role.

Reviewer: I found out that an empty string value yields inconsistent results:

- `ansible_host=` is put into the config file, causing the value not get updated on upload.
- The Client is called with `--ansible_host=`, removing the value from the Inventory.

Even though this “empty value” refers to the YAML syntax `value:`, it sounds ambiguous to me. I’d rather not describe features of YAML and be more explicit with empty strings.

```suggestion
* `null`: the ansible host name is not changed.
```

if we wanted to use an empty string to reset the value, which I don’t prefer, or

```suggestion
* `null` or an empty string: the ansible host name is not changed.
```

if we chose to consider an empty string an empty value, equivalent to `null` or `omit`.

It may be worth mentioning `omit` too.

```suggestion
* `null`, `omit`, or an empty string: the ansible host name is not changed.
```

> * `{state: absent}`: the ansible host name is unset in the insights-client config file and Host Based Inventory (HBI) is updated to use the system host name.

Reviewer: ```suggestion
* `{state: absent}`: the ansible host name is unset in the Insights Client config file and Host Based Inventory (HBI) is updated to use the system host name.
```

or

```suggestion
* `{state: absent}`: the ansible host name is unset in the insights-client.conf file and Host Based Inventory (HBI) is updated to use the system host name.
```

> * `{state: absent}`: the ansible host name is unset in the insights-client config file and Host Based Inventory (HBI) is updated to use the system host name.

Reviewer: If we decided that an empty string means the same as `{state: absent}`:

```suggestion
* `{state: absent}` or an empty string: the ansible host name is unset in the insights-client config file and Host Based Inventory (HBI) is updated to use the system host name.
```

I don’t prefer this solution though.

Reviewer 2: For some history about why we use the seemingly unintuitive `{state: absent}` - see https://linux-system-roles.github.io/documentation/incremental_settings.html

Also, our usage of `omit` is problematic - `omit` should almost never be used except for as a direct argument to a module parameter - users should certainly not be setting role parameters to the value `omit` - I would prefer not to mention it.

> * any other string value: the ansible host name is changed in Host Based Inventory (HBI).

Reviewer: ```suggestion
* any other string value: the Ansible host name is changed in Host Based Inventory (HBI).
```

> * any other string value: the ansible host name is changed in Host Based Inventory (HBI).

Reviewer: This is not true: an empty string value `""` removes the Ansible hostname value from the Inventory. The word “other” suggests that the “empty value” above may refer to an ampty string too, but that’s not true either and yields inconsistent results.

I‘d vote for removing the ambiguous wording “empty value” as `value:` is equivalent to `value: null` in YAML. See the conversation above.

```suggestion
* any non-empty string value: the ansible host name is changed in Host Based Inventory (HBI).
```

Whether the system automatically updates the dynamic configuration. It is
enabled by default.

```yaml
    rhc_insights:

>       ansible_host: null

Reviewer 3: like the other examples that set string values, I'd specify an example of text/string

```

Configures the ansible host name with a custom value for the system record 
in Host Based Inventory (HBI). This host name is used in playbooks by remediations.
It is `null` by default and will use the system host name if not specified.

Possible values of this variable:

> * `null` or an empty value: the ansible host name is not changed from the system host name.

Reviewer 4: I think this could be a bit confusing.
If you have previously set a ansible host name, and run a new playbook with this `null` parameter. I will expect the name to change to the system host name.

```suggestion
* `null` or an empty value: the ansible host name is not changed.
```

Author: Yeah this makes sense to me, thanks!
